apps/web/src/pages/integrations/components/Modal/ConnectIntegrationForm.tsx

```diff
@@ -0,0 +1,379 @@
+import { Controller, useForm } from 'react-hook-form';
+import styled from '@emotion/styled/macro';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useClipboard } from '@mantine/hooks';
+import { Image, useMantineColorScheme, Stack, Alert, ActionIcon, Center } from '@mantine/core';
+import { WarningOutlined } from '@ant-design/icons';
+import { ChannelTypeEnum, ICredentialsDto, IConfigCredentials } from '@novu/shared';
+
+import { Button, colors, Input, shadows, Switch, Text } from '../../../../design-system';
+import { IIntegratedProvider } from '../../IntegrationsStorePage';
+import { createIntegration, getWebhookSupportStatus, updateIntegration } from '../../../../api/integration';
+import { Close } from '../../../../design-system/icons/actions/Close';
+import { IntegrationInput } from '../IntegrationInput';
+import { API_ROOT } from '../../../../config';
+import { useEnvController, useAuthController } from '../../../../hooks';
+import { Check, Copy } from '../../../../design-system/icons';
+import { CONTEXT_PATH } from '../../../../config';
+import { successMessage } from '../../../../utils/notifications';
+
+enum ACTION_TYPE_ENUM {
+  HANDLE_SHOW_SWITCH = 'handle_show_switch',
+  HANDLE_ALERT_AND_ERROR_MSG = 'handle_alert_and_error_msg',
+  TOGGLE_CHECK = 'toggle_check',
+}
+
+type ActionType =
+  | { type: ACTION_TYPE_ENUM.TOGGLE_CHECK }
+  | {
+      type: ACTION_TYPE_ENUM.HANDLE_SHOW_SWITCH;
+      payload: boolean;
+    }
+  | {
+      type: ACTION_TYPE_ENUM.HANDLE_ALERT_AND_ERROR_MSG;
+      payload: {
+        isShowAlert: boolean;
+        errorMsg: string;
+      };
+    };
+
+const checkIntegrationInitialState = {
+  check: true,
+  isShowSwitch: false,
+  isShowAlert: false,
+  errorMsg: '',
+};
+
+const checkIntegrationReducer = (state: typeof checkIntegrationInitialState, action: ActionType) => {
+  switch (action.type) {
+    case ACTION_TYPE_ENUM.HANDLE_SHOW_SWITCH:
+      return {
+        ...state,
+        isShowSwitch: action.payload,
+      };
+
+    case ACTION_TYPE_ENUM.HANDLE_ALERT_AND_ERROR_MSG:
+      return {
+        ...state,
+        isShowAlert: action.payload.isShowAlert,
+        errorMsg: action.payload.errorMsg,
+      };
+
+    case ACTION_TYPE_ENUM.TOGGLE_CHECK:
+      return {
+        ...state,
+        check: !state.check,
+      };
+
+    default:
+      return state;
+  }
+};
+
+export function ConnectIntegrationForm({
+  provider,
+  showForm,
+  createModel,
+  onClose,
+  closeIntegration,
+}: {
+  provider: IIntegratedProvider | null;
+  showForm: (visible: boolean) => void;
+  createModel: boolean;
+  onClose: () => void;
+  closeIntegration: () => void;
+}) {
+  const {
+    register,
+    handleSubmit: handleSubmitIntegration,
+    setValue,
+    formState: { errors },
+    control,
+  } = useForm({ shouldUseNativeValidation: false });
+
+  const { colorScheme } = useMantineColorScheme();
+  const [isActive, setIsActive] = useState<boolean>(!!provider?.active);
```

Comment:
one switch state you have in the `useReducer` and another here, please move to the reducer then